### ISSUE TITLE
Cannot reproduce Decoder training (matching/better rFID, but worse gFID)

### ISSUE THREAD

**Issue by BartekPog** [bytetriper/RAE#80]
Thanks for releasing the code and checkpoints, great work!

I was trying to reproduce the Stage-1 ViT-XL decoder (DINOv2-B, τ=0.8) **from scratch** using the released recipe, and consistently end up with a decoder that **reconstructs as well or better (rFID) but generates noticeably worse (gFID)** than the released decoder. I'd really appreciate any pointers on training details I might be missing.

## Setup 
(following the repo as closely as I can)

- **Config:** `configs/stage1/training/DINOv2-B_decXL.yaml`, unmodified: 16 epochs, from scratch, lr 2e-4 cosine (1-epoch warmup), GAN on at epoch 8 (`disc_upd_start: 6`, `disc_start: 8`), DINO-S/8 discriminator, `noise_tau: 0.8`, L1 + LPIPS + adaptive GAN.
- **Command:**
  ```bash
  torchrun --nproc_per_node=8 src/train_stage1.py \
    --config configs/stage1/training/DINOv2-B_decXL.yaml \
    --data-path <imagenet_train> --image-size 256 --precision fp32
  ```
- **Hardware / data:** 8× H200, global batch 512 (64/GPU), full ImageNet-1k train (verified: 1000 classes, 1,281,167 images, uncorrupted). I trained both `bf16` and `fp32` (per the README's fp32 recommendation).

## rFID-50k (ImageNet val) is the same if not slightly better

| DINOv2-B (τ=0.8) | reported | README "reproduced" | released ckpt (my eval) | **my decoder (fp32, 16ep)** |
|---|---|---|---|---|
| rFID-50k | 0.57 | 0.54 | 0.569 | **0.535** |

So reconstruction matches the paper, my from-scratch decoder even has slightly *lower* rFID than the released checkpoint.

## gFID-50k is consistently worse

Decoding the **same** generated latents (from the released `DiTDH-XL` DINOv2-B model) with the released decoder vs. my reproduced decoder. Sampler: ODE Euler, 50 steps, `--label-sampling equal`; guidance from the shipped `DiTDHXL-DINOv2-B{,_AG}.yaml` configs.

| DiT / guidance | released decoder | **My decoder (fp32)** | Δ |
|---|---|---|---|
| DiTDH-XL, no guidance (cfg=1.0) | 1.55 | **1.70** | +0.15 |
| DiTDH-XL, autoguidance (1.42) | 1.155 | **1.22** | +0.06 |
| DiTDH-S, no guidance | 10.16 | **10.46** | +0.30 |
| DiTDH-S, cfg=1.5 | 7.74 | **8.17** | +0.43 |

The gap is consistent across DiTs and guidance settings.

## My evaluation is validated

To rule out an eval/metric problem, decoding with the **released** decoder reproduces the paper's numbers on my pipeline (gFID **1.55** no-guidance / **1.155** autoguidance ≈ reported 1.51 / 1.13), and my FID matches the ADM `evaluator.py` (TF InceptionV3) to <0.01 on the same samples. So the ~0.15 gFID gap is attributable to the **decoder I trained**, not the sampling or metric.

## What I've already ruled out

- **Dataset:** full ImageNet-1k train (correct count, uncorrupted, correct RGB channel order and value range/normalization).
- **Precision:** `bf16` and `fp32` both land at ~1.70 XL no-guidance (fp32 marginally better, not enough).
- **From-scratch vs. finetuning the released decoder:** both converge to ~1.70 gFID.
- **Eval pipeline:** validated via the released decoder (above).

## Question

Is there a Stage-1 training detail not captured in `DINOv2-B_decXL.yaml` / the README that produces the released decoder's gFID? Specifically:

1. Was the released decoder trained for **more than 16 epochs / more steps** than the shipped config?
2. Any **EMA**, **discriminator schedule**, or **DiffAug** (eg. the blurring warmup that is implemented but not used) settings beyond the config?
3. Is the released `decoders/dinov2/wReg_base/ViTXL_n08/model.pt` from the **same setup** as defined in `DINOv2-B_decXL.yaml`, or from a longer / differently-tuned run?

Any guidance on reproducing that behavior would be really helpful! 
Thanks a lot!

## Environment

PyTorch 2.12.1+cu130, Python 3.13, CUDA 13.0, 8× NVIDIA H200, global batch 512.
